src/index.ts

```diff
@@ -280,3 +328,37 @@ export const getLogs = async (
   });
   return txDetails?.meta?.logMessages || [];
 };
+
+// Was getSimulationUnits
+// Credit https://twitter.com/stegabob, originally from
+// https://x.com/stegaBOB/status/1766662289392889920
+export const getSimulationComputeUnits = async (
+  connection: Connection,
+  instructions: Array<TransactionInstruction>,
+  payer: PublicKey,
+  lookupTables: Array<AddressLookupTableAccount> | [],
+): Promise<number | null> => {
+  const testInstructions = [
+    // Set an arbitrarily high number in simulation
+    // so we can be sure the transaction will succeed
+    // and get the real compute units used
+    ComputeBudgetProgram.setComputeUnitLimit({ units: 1_400_000 }),
+    ...instructions,
+  ];
+
+  const testTransaction = new VersionedTransaction(
+    new TransactionMessage({
+      instructions: testInstructions,
+      payerKey: payer,
+      recentBlockhash: PublicKey.default.toString(),
```

Comment:
This is a bit odd - I imagine in the simulated transaction the recent blockhash isn't actually tested, so we're just throwing in a stub value (in this case `11111111111111111111111111111111`). Does that sound accurate?

Reply:
Have you tested this? It may be important to just take in a blockhash

Reply:
- Have I tested that this code works? Yes.
 - Have I tested the theory above (that the recent blockhash isn't actually tested and that we're throwing in a stub value? I thought I'd ask first. testing it now though, though - by throwing in `Keypair.generate().publicKey.toString(),` as recentBlockhash - would seem to support the theory.

Reply:
We use this in prod, so I can confirm that it does indeed work. Normally (based on the docs at least), the blockhash is checked. However, the simulation request includes `replaceRecentBlockhash: true` in the options which should ignore any blockhash checks. I haven't actually checked if it even needs a valid stub value either though to be honest. I'll try and double check that later today and report back!